common/persistence/sql/sqlTaskManager.go

```diff
@@ -309,7 +309,7 @@ func (m *sqlTaskManager) ListTaskList(
 			TaskType: lastRow.TaskType,
 		})
 	case pageToken.ShardID+1 < m.nShards:
-		nextPageToken, err = gobSerialize(&taskListPageToken{ShardID: pageToken.ShardID + 1, TaskType: math.MinInt16})
+		nextPageToken, err = gobSerialize(&taskListPageToken{ShardID: pageToken.ShardID + 1, TaskType: math.MinInt16, DomainID: minUUID})
```

Comment:
also should use empty here instead of minUUID.

Reply:
The problem with that change is here it's storing a string, not a struct. I either need a min value that's a string, or I'd need to refactor the struct to store []byte which could have ripple effects.

Reply:
Looking at the code that parses UUIDs, it seems to me like the "000..." string would parse to an array of zeroes, which should actually be a minimum.

Reply:
That's a valid concern. 

To be backward-compatible, a solution is to add a wrapper function in this file for
line 283 `domainID := serialization.MustParseUUID(pageToken.DomainID)`
to return empty striing, if `pageToken.DomainID` is equal to minUUID.

Reply:
Alternatively, I think it's fine if you change the field to `[]byte`. Serialization will fail the activity and then workflow, and after restarting, it will be back. 

I am fine with either way.

Reply:
alternatively, we could use another field instead of change the field to []byte. 
For example `DomainUUID`. The existing pagination token will become as empty.

Reply:
>"000..." string would parse to an array of zeroes, which should actually be a minimum.

I was under the same impression too. but this issue seems saying that's not true.
https://github.com/uber/cadence/issues/3911
The only explanation is that MySQL did something else when converting a string into binary. 

@yux0 do you have more context about this?

Reply:
@kraney how about we add a unit test for `uuid.MustParseUUID`? by comparing that minUUID to 1000 random UUID. That's should be the simplies way to verify your assumption.

Reply:
@kraney I am not sure that assumption is valid as MySQL is still a black box there. I tried to find something to validate it but didn't find enough information to confirm. 

If that becomes too much for you(since you have done a lot), you can open another ticket like "Potential bug: Tasklist Scavenger may not scan all domains"  and we can tackle it later.

Reply:
An all-zero UUID is not strictly the minimum possible value - as you showed an empty UUID is smaller. Maybe that's the issue? But it does seem to be the practical minimum given the actual usage.

Reply:
I am not sure either. But I did another look at the code, it shouldn't cause any issue if you want to change the field from string to []byte. This long running activity doesn't store any progress in heartbeat details. So when a worker restart, the activity will have to start from very beginning.